client/app/components/queries/ScheduleDialog.jsx

```diff
@@ -19,6 +19,42 @@ const DATE_FORMAT = 'YYYY-MM-DD';
 const HOUR_FORMAT = 'HH:mm';
 const { Option, OptGroup } = Select;
 
+function TimeEditor(props) {
+  const [time, setTime] = useState(props.defaultValue);
+  const showUtc = time && !time.isUTC();
+
+  function onChange(newTime) {
+    setTime(newTime);
+    props.onChange(newTime);
+  }
+
+  return (
+    <React.Fragment>
+      <TimePicker
+        allowEmpty={false}
+        value={time}
+        format={HOUR_FORMAT}
+        minuteStep={5}
+        onChange={onChange}
+      />
+      {showUtc && (
+        <span className="utc">
+          ({ moment.utc(time).format(HOUR_FORMAT) } UTC)
```

Comment:
Why not to keep taking it from `newSchedule`?

Reply:
Cause I think it's wrong for a component to be fed this kind of data from a prop when it's actually just a different presentation derived from it's own state.

When it was a method `getTimeSelector()` it was ok to be coupled with it's own `newSchedule` implementation but now that it's an independent component, this coupling should be avoided.

That's how I see it.

Reply:
While the logic for converting to utc is short and trivial, it has the risk of introducing a bug where we will save one thing and show a different one. We should either have a function that used in both places or have a test that ensures the value we show is the value we store (the function is probably better and easier). 

But I'm not sure you should apply different rules to this component from the ones you used to the `getTimeSelector` function. They serve the same purpose: a helper to render the time selector and encapsulate some small logic about it. Both not supposed to be used beyond the context of `SchedueDialog`.

Reply:
Then I think we both agree any implementation has risk so testing is a must.
You ok I leave as is and add tests?

Reply:
Sure.

Reply:
Added following tests:
1. UTC set correctly on init (2h below set time)
2. UTC time should not render (if already UTC)
3. onChange correct result - simulates time change then asserts UTC and callback value

🏄🏄‍♀️